packages/loctool/lib/ResourceString.js

```diff
@@ -44,6 +44,7 @@ var ResourceString = function(props) {
     if (props) {
         this.source = typeof(props.source) === 'string' ? props.source : props.text;
         this.target = props.target;
+        this.metadata = props.metadata || undefined;
```

Comment:
Move these changes up to Resource.js instead of ResourceString.js so that it applies to arrays and plurals too. Also, add some tests for arrays and plurals. (Same test as below, but for the other subtypes)

Reply:
I’ve updated it and applied the same change to ilib‑tools‑common.